### ISSUE TITLE
`seed` value is not used by many splitters

### ISSUE THREAD

**Issue by gusty1g** [deepchem/deepchem#3228]
Many splitters are not utilizing the `seed` argument passed to the `split` method. Some splitters like the `IndexSplitter` do not require `seed` attribute. Can we deprecate `seed` argument in such cases?

Can we make `seed` as a class attribute? Because I think `seed` is the property of the `Splitter` class rather than the method - it is used by multiple methods in the `Splitter` method like `train_test_valid_split`, `k_fold_split` as well as `train_test_split`.

Related to #3219

**Comment by rbharath** [comment]
`seed` is only used if randomness is needed by the splitter. I think for cases like `IndexSplitter`, it's kept to retain compatibility with the superclass.

It should not be a class attribute. Different splitter instances may have different sources of randomness. It would be dangerous to have one class attribute controlling all splitter instances in parallel. It could become an instance attribute potentially though, with each instance passed a `seed` in its constructor. This would require a deprecation cycle and I'm not sure this is worth the effort but I'm open to be convinced otherwise
